### PR TITLE
2 category model

### LINKED ISSUE #2: Category model
Create a `Category` model to represent item categories.

### PR DESCRIPTION
**Pull request by josephbridgwaterrowe**
Closes #2

### CODE REVIEW

**Comment by josephbridgwaterrowe** [comment]
@michaelachrisco please review.

File: app/helpers/fagus/application_helper.rb [hunk 1]
```diff
@@ -1,4 +1,19 @@
+# Fagus
 module Fagus
+  # ApplicationHelper
   module ApplicationHelper
+    # A simple way to show the error messages for an ActiveRecord model.
+    # @params object [Object] the object inheriting from ActiveRecord::Base.
+    # @return [String] the error messages in HTML.
+    def error_messages(object)
+      return nil if object.errors.nil? || object.errors.empty?
```
Comment (michaelachrisco) [inline app/helpers/fagus/application_helper.rb]: You can rewrite this as:

``` ruby
def error_messages(object)
  return nil if object.blank?
  ...
end
```

File: app/helpers/fagus/categories_helper.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+module Fagus
```
Comment (michaelachrisco) [inline app/helpers/fagus/categories_helper.rb]: Probably want to remove unneeded helpers.

File: app/models/fagus/category.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+module Fagus
+  class Category < ActiveRecord::Base
+    extend Enumerize
```
Comment (michaelachrisco) [inline app/models/fagus/category.rb]: Are you using Enumerize on any of the attributes?

File: app/views/fagus/categories/index.json.jbuilder [hunk 4]
```diff
@@ -0,0 +1,4 @@
+json.array! @categories do |category|
+  json.id category.id
+  json.name category.name
```
Comment (michaelachrisco) [inline app/views/fagus/categories/index.json.jbuilder:3]: I believe you need to add in `category.reference` if you are planning on using jquery.

Comment (michaelachrisco) [inline app/views/fagus/categories/index.json.jbuilder:3]: As well as `category.description` and  `category.uuid`.

File: spec/factories/categories.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+FactoryGirl.define do
+  factory :category, :class => Fagus::Category do
+    description { "#{reference} - #{name}" }
```
Comment (michaelachrisco) [inline spec/factories/categories.rb:3]: You have `is_active` declared in the schema. You will probably need to add the field into the factory.

File: spec/features/user_updates_an_existing_category_spec.rb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+feature 'User updates an existing category' do
+  given(:category) { create(:category) }
+
+  context 'when the details are valid' do
+    scenario 'the user sees a success message' do
+      visit edit_category_path(category)
+
+      fill_in :category_name, :with => Faker::Lorem.word
+      fill_in :category_description, :with => Faker::Lorem.word
+      # fill_in :category_reference, :with => Faker::Number.number(3)
```
Comment (michaelachrisco) [inline spec/features/user_updates_an_existing_category_spec.rb:12]: Do you need this reference?

File: spec/dummy/log/production.log [hunk 7]
```diff
@@ -0,0 +1,8 @@
+D, [2015-07-02T15:45:47.888437 #86981] DEBUG -- :   [1m[36m (54.5ms)[0m  [1mCREATE TABLE `schema_migrations` (`version` varchar(255) NOT NULL) ENGINE=InnoDB[0m
```
Comment (michaelachrisco) [inline spec/dummy/log/production.log:1]: I believe you need to remove this file and add it to the `.gitignore`

**Comment by michaelachrisco** [comment]
I dont see you using Enumerize anywhere in the models. Is this gem used somewhere else?

File: app/helpers/fagus/application_helper.rb (hunk 1 above)
Comment (josephbridgwaterrowe) [inline app/helpers/fagus/application_helper.rb]: Thanks, but do you mean object.errors.blank?

File: app/models/fagus/category.rb [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+    after_initialize :ensure_uuid_present
+
+    validates \
+      :description,
+      :name,
+      :reference,
+      :uuid, :presence => true
+
```
Comment (michaelachrisco) [inline app/models/fagus/category.rb:10]: I believe that you can add in 

``` ruby
  def active?
    is_active == 1
  end
```

as we have used in other projects.

File: app/helpers/fagus/categories_helper.rb (hunk 2 above)
Comment (josephbridgwaterrowe) [inline app/helpers/fagus/categories_helper.rb]: Ta, got it.

File: app/helpers/fagus/application_helper.rb (hunk 1 above)
Comment (michaelachrisco) [inline app/helpers/fagus/application_helper.rb]: Yes. Thanks.

File: app/models/fagus/category.rb (hunk 3 above)
Comment (josephbridgwaterrowe) [inline app/models/fagus/category.rb]: No, good catch... :baby:

File: app/views/fagus/categories/index.json.jbuilder (hunk 4 above)
Comment (josephbridgwaterrowe) [inline app/views/fagus/categories/index.json.jbuilder:3]: I'm not at the moment, I just left this is as it was a copy and paste and I didn't want to have to look it up as I would probably forget the jbuilder syntax by the time its needed :smile:

File: spec/factories/categories.rb (hunk 5 above)
Comment (josephbridgwaterrowe) [inline spec/factories/categories.rb:3]: It defaults to `1` but I will add for clarify. Ta.

File: spec/features/user_updates_an_existing_category_spec.rb (hunk 6 above)
Comment (josephbridgwaterrowe) [inline spec/features/user_updates_an_existing_category_spec.rb:12]: I wasn't sure if it should be ready only or not.

File: spec/dummy/log/production.log (hunk 7 above)
Comment (josephbridgwaterrowe) [inline spec/dummy/log/production.log:1]: Whoops, yes.

**Comment by josephbridgwaterrowe** [comment]
Removing Enumerize as its not used at the moment.

File: app/views/fagus/categories/show.html.erb [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+  <%= @category.name %>
+</div>
+<div>
+  <label>Description</label>
+  <%= @category.description %>
+</div>
+<div>
+  <label>Reference</label>
+  <%= @category.reference %>
+</div>
```
Comment (michaelachrisco) [inline app/views/fagus/categories/show.html.erb:21]: I believe you need to add in `is_active` in this view.

File: app/views/fagus/categories/index.json.jbuilder (hunk 4 above)
Comment (michaelachrisco) [inline app/views/fagus/categories/index.json.jbuilder:3]: In that case, you should probably remove the file altogether and add this in later if needed.

**Comment by josephbridgwaterrowe** [comment]
@michaelachrisco changes made.

**Comment by michaelachrisco** [comment]
Ok left a few comments.

**Comment by michaelachrisco** [comment]
@josephbridgwaterrowe Rebase and I will merge.

**Comment by josephbridgwaterrowe** [comment]
@michaelachrisco rebased.

**Comment by michaelachrisco** [comment]
Cool beans :+1:

**Pull request merged**